.pipelines/templates/e2e-test-kind-arc.yaml

```diff
@@ -0,0 +1,43 @@
+jobs:
+  - job: e2e_arc_kind
+    variables:
+    - group: csi-secrets-store-e2e
+    steps:
+    - script: |
+        az login -i > /dev/null
+        az account set -s=$(SUBSCRIPTION_ID)
+      displayName: "az login"
+    - script: |
+        az extension add --name connectedk8s
+        az extension add --name k8s-extension
+      displayName: "add cli extensions"
+    - script: |
+        make install-kubectl setup-kind
+      displayName: "setup KinD"
+    - script: |
+        clusterName=sscd-arc-e2e-$(openssl rand -hex 6)
+        echo "##vso[task.setvariable variable=AZURE_CLUSTER_NAME]clusterName"
+        echo "Cluster name is set to - $clusterName"
+        echo ${AZURE_CLUSTER_NAME}
+      displayName: "set cluster name"
+      condition: succeeded()
+    - script: |
+        az group create -n ${AZURE_CLUSTER_NAME} -l $(AZURE_CANARY_LOCATION)
+
+        az connectedk8s connect -n ${AZURE_CLUSTER_NAME} -g ${AZURE_CLUSTER_NAME} --no-wait
```

Comment:
why are we using no wait here but then loop to check connection later?

Reply:
It takes time for Arc pods to come up. Sometimes, in such cases helm might report [unable to install helm release](https://dev.azure.com/AzureContainerUpstream/Secrets%20Store%20CSI%20Driver%20Provider%20Azure/_build/results?buildId=28550&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=c0307c8e-cb29-5919-a192-498b9fd8bfd8&l=28), but in fact Arc operators gets installed and able to connect cluster. 

Also, `az connectedk8s connect` will go through [different phases](https://dev.azure.com/AzureContainerUpstream/Secrets%20Store%20CSI%20Driver%20Provider%20Azure/_build/results?buildId=28551&view=logs&j=b2320112-bec5-5b88-4bdd-c7b41e94b892&t=86772953-b9fc-52e3-20eb-8309e7dd6de0&l=28) of installation. So to address both, we are checking the status later without waiting.

We are same doing for extension install.

Reply:
Could you add this as a comment in the yaml? It'll be good for future reference.

Reply:
added